setup.cfg

```diff
@@ -32,13 +32,19 @@ install_requires =
     pystac[validation] ~= 1.0.0
     aiohttp ~= 3.7
     click ~= 7.1
-    fsspec ~= 2021.6.0
+    fsspec ~= 2021.7
     requests ~= 2.25
     Shapely ~= 1.7
     pyproj ~= 3.0
     rasterio ~= 1.2
     lxml ~= 4.6
 
+[options.extras_require]
+all =
+    %(s3)s
+s3 =
+    s3fs ~= 2021.7
+
```

Comment:
@gadomski Are you OK with this?

Reply:
Can you explain what it's doing? Is it expanding the `s3` variable to its string value? Then -- maybe? I guess I'm not convinced its worth it to do backflips to enable an `setuptools[all]` quite yet. My impression is most users using non-standard features (e.g. if we enabled Azure Blob storage URLs or something like that) would be working a bit off the beaten path.

Reply:
Yeah, I'm not enthralled with the level of non-standard cleverness. But if we end up adding more it might be useful.
It's using `configparser`'s [BasicInterpolation](https://docs.python.org/3/library/configparser.html#configparser.BasicInterpolation).

Reply:
Ok, I'm good w/ keeping it. Merge away! Thanks for the updates!